src/extension.ts

```diff
@@ -57,13 +101,39 @@ async function fetchPackageMetadata(requirement: ProjectNameRequirement): Promis
     return await metadataCache.get(requirement.name)!()
 }
 
+let lastLine = -1
+let lastHover: vscode.Hover | null = null
+let lastLineText = ''
+
 class PyPIHoverProvider implements vscode.HoverProvider {
     async provideHover(document: vscode.TextDocument, position: vscode.Position): Promise<vscode.Hover | null> {
-        const requirement = parsePipRequirementsLineLoosely(document.lineAt(position.line).text)
+        let lineText = document.lineAt(position.line).text
+        if (lastLine === position.line && lineText === lastLineText) {
+            return lastHover
+        }
+
+        const isPyProjectToml =
+            document.languageId === 'toml' && document.fileName.toLowerCase().endsWith('pyproject.toml')
+
+        if (isPyProjectToml) {
+            const dependencies = getPyProjectDependencies(document.getText())
+            // Workaround for extracting just the package name from the line
+            lineText = lineText.split('=')[0].trim()
```

Comment:
I think it wouldn't be _too_ complicated to avoid the workaround by using an AST parser (e.g. https://github.com/ota-meshi/toml-eslint-parser) that'd allow us to map lines to individual dependencies – but the current solution is totally fine